src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMHead.java

```diff
@@ -87,9 +87,15 @@
         this.sourceOwner = repository == null ? null : repository.getOwnerName();
         this.sourceRepo = repository == null ? null : repository.getName();
         this.sourceBranch = pr.getHead().getRef();
-        this.origin = pr.getRepository().getOwnerName().equalsIgnoreCase(sourceOwner)
-                ? SCMHeadOrigin.DEFAULT
-                : new SCMHeadOrigin.Fork(this.sourceOwner);
+
+        if (pr.getRepository().getOwnerName().equalsIgnoreCase(sourceOwner)) {
+            this.origin = SCMHeadOrigin.DEFAULT;
+        } else {
+            // if the forked repo name differs from the upstream repo name
+            this.origin = pr.getHead().getRepository().getName().equalsIgnoreCase(repoName)
```

Comment:
I think you can do this without changing the constructor and so on:
```suggestion
            this.origin = pr.getHead().getRepository().getName().equalsIgnoreCase(pr.getBase().getRepository().getName())
```

If this works you can rollback the changes out calling code and constructor.